java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/LspServerTelemetryManager.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.lsp.server.protocol;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.WeakHashMap;
+import java.util.concurrent.Future;
+import org.eclipse.lsp4j.MessageType;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.netbeans.api.java.queries.CompilerOptionsQuery;
+import org.netbeans.api.java.queries.CompilerOptionsQuery.Result;
+import org.netbeans.api.project.Project;
+import org.netbeans.api.project.ui.ProjectProblems;
+import org.openide.filesystems.FileObject;
+import org.openide.util.Exceptions;
+
+/**
+ *
+ * @author atalati
+ */
+public class LspServerTelemetryManager {
+
+    public final String SCAN_START_EVT = "SCAN_START_EVT";
+    public final String SCAN_END_EVT = "SCAN_END_EVT";
+    public final String WORKSPACE_INFO_EVT = "WORKSPACE_INFO_EVT";
+
+    private final String ENABLE_PREVIEW = "--enable-preview";
+    private final String STANDALONE_PRJ = "Standalone";
+    private final WeakHashMap<LanguageClient, Future<Void>> clients = new WeakHashMap<>();
+    private long lspServerIntiailizationTime;
+
+    public synchronized void connect(LanguageClient client, Future<Void> future) {
+        clients.put(client, future);
+        lspServerIntiailizationTime = System.currentTimeMillis();
+    }
+
+    public void sendTelemetry(TelemetryEvent event) {
```

Comment:
This method is accessed from multiple threads; synchronize access to shared variables.